multiversx_sdk_cli/cli_shared.py

```diff
@@ -172,6 +176,28 @@ def prepare_nonce_in_args(args: Any):
         args.nonce = account.nonce
 
 
+def prepare_chain_id_in_args(args: Any):
+    if not args.chain and not args.proxy:
+        raise ArgumentsNotProvidedError("chain ID cannot be decided: `--chain` or `--proxy` should be provided")
+
+    if args.chain and args.proxy:
+        proxy = ProxyNetworkProvider(args.proxy)
+        fetched_chain_id = proxy.get_network_config().chain_id
+
+        if args.chain != fetched_chain_id:
+            show_warning("The chain ID you have provided does not match the chain ID you got from the proxy. Will use the value from the proxy.")
```

Comment:
does not match the chain ID retrieved from the proxy. Will use the proxy's value (%s)

where %s is the actual chain id

Reply:
Done!